cvasl/seperated.py

```diff
@@ -537,3 +550,200 @@ def check_sex_dimorph_expectations(dataframe):
     else:
         bad_data = 0
     return bad_data
+
+
+def avg_k_folds(frame):
+    """
+    This function takes a dataframe of k_fold results,
+    as formatted for our experiments with derived datasets
+    and returns an averaged dataframe
+    """
+    data = [[
+        frame['algorithm'][0],
+        frame['file_name'][0],
+        frame.mae.mean(),
+        frame.r2.mean(),
+        frame.explained_variance.mean()]]
+    frame_results_average = pd.DataFrame(
+        data,
+        columns=['algorithm', 'file_name', 'mae', 'r2', 'explained_variance']
+    )
+    return frame_results_average
+
+
+def frame_a_model_sex_split(
+        model_name,
+        model_file_name,
+        scikit_model,
+        our_ml_matrix,
```

Comment:
```suggestion
        ml_matrix,
```

Reply:
`our_ml_matrix` is used only to retrieve the values from the "split" column (in your user case, 'sex'); I'd suggest either to give to the user the possibility of choosing which column to use (adding an input variable like `col_split` or similar) or to pass in the "split" column values only, instead of the entire matrix.